src/components/AdjustVaultForm.tsx

```diff
@@ -49,6 +58,67 @@ const AdjustVaultForm = () => {
 
   const isActive = vaultToAdjust?.vaultState === 'active';
 
+  const onMaxCollateralClicked = () => {
+    assert(collateralAction === CollateralAction.Deposit);
```

Comment:
helpful assertion. makes clear that when`actionLabel` is undefined the user can't click.